bee-crypto/src/ternary/sponge/batched_curlp/mod.rs

```diff
@@ -44,7 +47,11 @@ pub struct BatchHasher {
     curlp: CurlP,
 }
 
-impl BatchHasher {
+impl<B, S> BatchHasher<B>
```

Comment:
You can have this just be generic over `B`. `RawEncodingBuf` has an associated type called `Slice` that gives you the slice version of that buffer.